gnucash/gnome-utils/gnc-autoclear.cpp

```diff
@@ -0,0 +1,198 @@
+/********************************************************************
+ * gnc-autoclear.c -- Knapsack algorithm functions                  *
+ *                                                                  *
+ * Copyright 2020 Cristian Klein <cristian@kleinlabs.eu>            *
+ * Modified  2021 Christopher Lam to clear same-amount splits       *
+ * Modified  2025 ChatGPT to merge without building a shifted vec   *
+ *                                                                  *
+ * This program is free software; you can redistribute it and/or    *
+ * modify it under the terms of the GNU General Public License as   *
+ * published by the Free Software Foundation; either version 2 of   *
+ * the License, or (at your option) any later version.              *
+ *                                                                  *
+ * This program is distributed in the hope that it will be useful,  *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of   *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the    *
+ * GNU General Public License for more details.                     *
+ *                                                                  *
+ * You should have received a copy of the GNU General Public License*
+ * along with this program; if not, contact:                        *
+ *                                                                  *
+ * Free Software Foundation           Voice:  +1-617-542-5942       *
+ * 51 Franklin Street, Fifth Floor    Fax:    +1-617-542-2652       *
+ * Boston, MA  02110-1301,  USA       gnu@gnu.org                   *
+ *******************************************************************/
+#include <config.h>
+
+#include <gtk/gtk.h>
+#include <glib/gi18n.h>
+
+#include "Account.h"
+#include "Account.hpp"
+#include "Split.h"
+#include "Transaction.h"
+#include "gnc-autoclear.h"
+
+#include <vector>
+#include <numeric>
+#include <algorithm>
+#include <cstdint>
+
+#define log_module "autoclear"
+
+using SplitVec = std::vector<Split*>;
+
+
+struct Entry
+{
+    int64_t sum;
+    int64_t pred_index;   // index of last split
+    bool unique;          // true if sum is still unique
+    bool operator<(const Entry& o) const { return sum < o.sum; }
+};
+
+/**
+ * subset_sum_unique
+ * @splits: list of split amounts
+ * @target: target sum
+ * @result: output indices of splits that sum to target if unique
+ * Returns nullptr on success, otherwise returns error message.
+ *
+ * This implementation merges the existing sums vector with a virtual
```

Comment:
It's not existing after this commit. History goes in git commit messages, comments should explain the current code.